### PR TITLE
Misc updates to the WGS pipelines

### PR DESCRIPTION
**Pull request by SHuang-Broad**
Learnt several things while processing a cohort of medium size.
Hence we encode them here.

### CODE REVIEW

**Review by kvg — APPROVED** [review]
Lots of good stuff here. There some stuff I'm not convinced by, but I'll comment more extensively in subsequent PRs.

File: docker/lr-dv/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM google/deepvariant:1.3.0
+
+COPY vm_local_monitoring_script.sh /opt/
```
Comment (kvg) [inline docker/lr-dv/Dockerfile:3]: Yeah, nice workaround, but wish we didn't need it.

File: docker/lr-dv/Makefile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+VERSION = 1.3.0
+TAG1 = us.gcr.io/broad-dsp-lrma/lr-deepvariant:$(VERSION)
```
Comment (kvg) [inline docker/lr-dv/Makefile:2]: Nice. We should make a plan to retire all of our quay.io uploads now. They're just doubling our Docker build times for no good reason.

File: wdl/ONTWholeGenome.wdl [hunk 3]
```diff
@@ -27,10 +27,10 @@ workflow ONTWholeGenome {
 
         String gcs_out_root_dir
 
-        Boolean call_variants = true
-
+        Boolean call_svs
```
Comment (kvg) [inline wdl/ONTWholeGenome.wdl:30]: In general, I think nearly all workflow "flags" like this should be set to a reasonable default, ideally whatever settings we would use if the data was our own. Users can override it if they choose, but should they only specify input files and nothing else, I think the workflow should still run properly.

File: wdl/ONTWholeGenome.wdl [hunk 4]
```diff
@@ -91,28 +91,44 @@ workflow ONTWholeGenome {
...
-        if (! defined(run_dv_pepper_analysis)) {call Utils.StopWorkflow as run_dv_pepper_analysis_not_provided {input: reason = "Unprovided arg run_dv_pepper_analysis"}}
-        if (! defined(dvp_threads)) {call Utils.StopWorkflow as dvp_threads_not_provided {input: reason = "Unprovided arg dvp_threads"}}
-        if (! defined(ref_scatter_interval_list_locator)) {call Utils.StopWorkflow as ref_scatter_interval_list_locator_not_provided {input: reason = "Unprovided arg ref_scatter_interval_list_locator"}}
-        if (! defined(ref_scatter_interval_list_ids)) {call Utils.StopWorkflow as ref_scatter_interval_list_ids_not_provided {input: reason = "Unprovided arg ref_scatter_interval_list_ids"}}
-        if (! defined(call_small_vars_on_mitochondria)) {call Utils.StopWorkflow as call_small_vars_on_mitochondria_not_provided {input: reason = "Unprovided arg call_small_vars_on_mitochondria"}}
+        if (call_svs) {
+            if (! defined(fast_less_sensitive_sv)) {call Utils.StopWorkflow as fast_less_sensitive_sv_not_provided {input: reason = "Calling SVs without specifying arg fast_less_sensitive_sv"}}
```
Comment (kvg) [inline wdl/ONTWholeGenome.wdl:106]: This is the major section of the PR that gives me pause. In trying to apply the workflows to malaria data, I found this section to cause issues.  The problem is that Terra doesn't have a way of telling you these optional parameters are actually required.  Only the failure of the workflow reveals the fact that values must be supplied for these parameters.

File: wdl/ONTWholeGenome.wdl [hunk 5]
```diff
@@ -91,28 +91,44 @@ workflow ONTWholeGenome {
...
+        }
+        if (call_small_variants) {
+            if (! defined(call_small_vars_on_mitochondria)) {call Utils.StopWorkflow as call_small_vars_on_mitochondria_not_provided {input: reason = "Unprovided arg call_small_vars_on_mitochondria"}}
+            if (! defined(run_dv_pepper_analysis)) {call Utils.StopWorkflow as run_dv_pepper_analysis_not_provided {input: reason = "Unprovided arg run_dv_pepper_analysis"}}
+            if (! defined(dvp_threads)) {call Utils.StopWorkflow as dvp_threads_not_provided {input: reason = "Unprovided arg dvp_threads"}}
```
Comment (kvg) [inline wdl/ONTWholeGenome.wdl:111]: Thematically related to my earlier comments about setting these to sensible defaults, parameters like this are very hard for the typical user to simply know.  Likely they'll supply something like "1" or "2", or something ridiculous like "1000".  We should simply set this to what we think it should be, and then users can change it if they so choose.

File: wdl/PBCCSWholeGenome.wdl [hunk 6]
```diff
@@ -85,28 +85,45 @@ workflow PBCCSWholeGenome {
...
-        if (! defined(run_dv_pepper_analysis)) {call Utils.StopWorkflow as run_dv_pepper_analysis_not_provided {input: reason = "Unprovided arg run_dv_pepper_analysis"}}
-        if (! defined(dvp_threads)) {call Utils.StopWorkflow as dvp_threads_not_provided {input: reason = "Unprovided arg dvp_threads"}}
-        if (! defined(ref_scatter_interval_list_locator)) {call Utils.StopWorkflow as ref_scatter_interval_list_locator_not_provided {input: reason = "Unprovided arg ref_scatter_interval_list_locator"}}
-        if (! defined(ref_scatter_interval_list_ids)) {call Utils.StopWorkflow as ref_scatter_interval_list_ids_not_provided {input: reason = "Unprovided arg ref_scatter_interval_list_ids"}}
-        if (! defined(call_small_vars_on_mitochondria)) {call Utils.StopWorkflow as call_small_vars_on_mitochondria_not_provided {input: reason = "Unprovided arg call_small_vars_on_mitochondria"}}
+        if (call_svs) {
+            if (! defined(fast_less_sensitive_sv)) {call Utils.StopWorkflow as fast_less_sensitive_sv_not_provided {input: reason = "Calling SVs without specifying arg fast_less_sensitive_sv"}}
```
Comment (kvg) [inline wdl/PBCCSWholeGenome.wdl:101]: See earlier comments on defaults.

File: wdl/PBCLRWholeGenome.wdl [hunk 7]
```diff
@@ -72,10 +71,21 @@ workflow PBCLRWholeGenome {
...
+
+    if (defined(bed_to_compute_coverage)) { call FF.FinalizeToFile as FinalizeRegionalCoverage { input: outdir = dir, file = select_first([coverage.bed_cov_summary]) } }
+
+    if (call_svs || call_small_variants) {
+
         # verify arguments are provided
-        if (! defined(fast_less_sensitive_sv)) {call Utils.StopWorkflow as fast_less_sensitive_sv_not_provided {input: reason = "Unprovided arg fast_less_sensitive_sv"}}
-        if (! defined(call_small_vars_on_mitochondria)) {call Utils.StopWorkflow as call_small_vars_on_mitochondria_not_provided {input: reason = "Unprovided arg call_small_vars_on_mitochondria"}}
+        if (call_svs) {
+            if (! defined(fast_less_sensitive_sv)) {call Utils.StopWorkflow as fast_less_sensitive_sv_not_provided {input: reason = "Calling SVs without specifying arg fast_less_sensitive_sv"}}
```
Comment (kvg) [inline wdl/PBCLRWholeGenome.wdl:86]: See earlier comments on defaults.

**Review by kvg — APPROVED** [review]
Overall looks very good.  There's some stuff I'm not quite sold on (mostly regarding emitting errors when an optional workflow parameter doesn't have a setting).  But I think we can discuss that in person and address it in a subsequent PR.

File: docker/lr-dv/Dockerfile (hunk 1 above)
Comment (SHuang-Broad) [reply docker/lr-dv/Dockerfile:3]: The Terra-as-a-service idea, after discussing with Berry, seems like a no-go.
I'm sure there are wish-list documents living somewhere, but I'd like to know what their concerns are for exposing some of these Cromwell parameters.

File: docker/lr-dv/Makefile (hunk 2 above)
Comment (SHuang-Broad) [reply docker/lr-dv/Makefile:2]: we need human-power... :wink:

File: wdl/ONTWholeGenome.wdl (hunk 3 above)
Comment (SHuang-Broad) [reply wdl/ONTWholeGenome.wdl:30]: Providing defaults making sense for our own analysis is definitely one way to go.

What's lacking, to me, is a true variant validator like [barkely](https://github.com/broadinstitute/barclay), and that should really come from the Cromwell Team

File: wdl/ONTWholeGenome.wdl (hunk 4 above)
Comment (SHuang-Broad) [reply wdl/ONTWholeGenome.wdl:106]: Now that it's a blocker, I'll update.
I'll default to a cheap-version (alignment + scatter-gather SV), and turn off small variants.
If one wants to turn on small variants, DV-pepper will off by-default as it's expensive, so just remember to turn that on as well if intended.

File: wdl/ONTWholeGenome.wdl (hunk 5 above)
Comment (SHuang-Broad) [reply wdl/ONTWholeGenome.wdl:111]: will update

**Comment by SHuang-Broad** [comment]
default value provided.
mitocondria var calling is turned off by default

**Pull request merged**